hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieCompactor.java

```diff
@@ -101,6 +104,12 @@ public static class Config implements Serializable {
     public String runningMode = null;
     @Parameter(names = {"--strategy", "-st"}, description = "Strategy Class", required = false)
     public String strategyClassName = LogFileSizeBasedCompactionStrategy.class.getName();
+    @Parameter(names = {"--job-max-processing-time-ms", "-mt"}, description = "Take effect when using --mode/-m execute or scheduleAndExecute. "
+        + "If maxProcessingTimeMs passed but compaction job is still unfinished, hoodie would consider this job as failed and relaunch.")
+    public long maxProcessingTimeMs = 0;
+    @Parameter(names = {"--retry-last-failed-compaction-job", "-rc"}, description = "Take effect when using --mode/-m execute or scheduleAndExecute. "
```

Comment:
Another choice is to always rollback the previous pending compaction before starting a new one, we only need to assure that the compaction can be finished between the compaction job scheduling interval.

Reply:
Yes. We just only choose a failed compaction instant here. It will been rolled back in subsequent executions `SparkRDDTableServiceClient#compact(String compactionInstantTime, boolean shouldComplete)`

Reply:
Got it, so do you still need this PR?

Reply:
Yes, we need it to process the failed inflight compaction plan which will never been re-run in default way.

Reply:
> the failed inflight compaction plan which will never been re-run

Can we fix that rollback by including the inflight compactions instead of introducing new config options?

Reply:
We didn't think of a better way as so far. As https://github.com/apache/hudi/pull/8944 mentioned, we don't  know the inflight instant is still executing or already finished. If we rollback the inflight instant first, it does not suitable for the concurrent execution.

Reply:
You can actually check the heartbeat of the instant, if the heartbeat timed out, we can roll it back.

Reply:
> the offline job timed out does not represent the heartbeat of the instant also timed out 

We can, by default there is a 1 minutes timeout threshold for each instant, if an inflight instant does not receive heartbeat for 1 minutes, we can deem the txn as dead.

Reply:
I got your meaning, you are right. I change it, thanks